lib/onEvent/message.js

```diff
@@ -2,35 +2,81 @@ const Kirbi = require('../../../../kirbi');
 
 //initialize AI
 if (Kirbi.Config.elizaEnabled && !Kirbi.Eliza) {
-	var Eliza = Kirbi.require('./extras/eliza');
+	let Eliza = Kirbi.require('./extras/eliza');
 	Kirbi.Eliza = new Eliza();
 	console.log('Eliza enabled.');
 	Kirbi.Eliza.memSize = 500;
 };
 
+function currentChannelHasHook (channelId) {
+	if (Kirbi.Auth.slack.webhooks[Kirbi.Slack.dataStore.getChannelById(channelId).name]) return true;
+	return false;
+};
+
 function slackMessageCb (output, msg, expires, delCalling) {
 	//we need to format the message output for the Slack API first...
-	let reformatted = {};
+	if (typeof output === 'object') {
+		let reformatted;
+		reformatted = {};
+		reformatted.attachments = [];
+		reformatted.attachments[0] = {};
+		reformatted.attachments[0].fallback = 'There was an error';
+		if (output.embed.color) reformatted.attachments[0].color = `#${output.embed.color}`;
+		if (output.embed.title) reformatted.attachments[0].title = output.embed.title;
+		if (output.embed.description) reformatted.attachments[0].text = output.embed.description;
+		if (typeof output.embed.image !== 'undefined') reformatted.attachments[0].image_url = output.embed.image.url;
+		if (typeof output.embed.thumbnail !== 'undefined') reformatted.attachments[0].thumb_url = output.embed.thumbnail.url;
+		if (output.embed.footer) reformatted.attachments[0].footer = output.embed.footer;
+		if (typeof output.embed.author !== 'undefined') {
+			if (output.embed.author.name) reformatted.attachments[0].author_name = output.embed.author.name;
+			if (output.embed.author.url) reformatted.attachments[0].author_link = output.embed.author.url;
+			if (output.embed.author.icon_url) reformatted.attachments[0].author_icon = output.embed.author.icon_url;
+		}
+		if (typeof output.embed.fields !== 'undefined') {
+			reformatted.attachments[0].fields = [];
+			output.embed.fields.forEach(field => {
+				reformatted.attachments[0].fields.push({
+					title: field.name,
+					value: field.value,
+					short: field.inline
+				});
+			});
+		}
+		if (output.reply) reformatted.attachments[0].pretext = `<@${output.reply}>`;
 
+		if (currentChannelHasHook(msg.channel)) {
+			//check config for webhook urls...
+			//if the message came from a channel that we have a webhook for,
+			//send the attachment to that url
+			let url = Kirbi.Auth.slack.webhooks[Kirbi.Slack.dataStore.getChannelById(msg.channel).name];
+			require('request').post({
+				uri: url,
+				json: true,
+				body: reformatted
+			});
+		};
+		return;
+	};
 
 
 	//then send it and interact with it based on the supplied flags...
-	if (expires) {
-		return msg.channel.send(output).then(message => message.delete(5000));
-	}
-	if (delCalling) {
-		return msg.channel.send(output).then(() => msg.delete());
-	}
+	//if (expires) {
```

Comment:
delete if not being used.

Reply:
leaving in since it needs to be fixed in the future. I'll add a TODO